src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/IHttpHeadersHandler.cs

```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Http
+{
+    public interface IHttpHeadersHandler
```

Comment:
An interface call per header might be more expensive than we want to pay. Do we really need this to be abstracted? I think a parser abstraction makes sense, but it seems to me like a separate abstraction for handling headers (one by one) might be an overkill.